### PR TITLE
[msquic] Rename from ms-quic and remove the tools feature.

### PR DESCRIPTION
**Pull request by BillyONeal**
As requested by upstream maintainer @nibanks in https://github.com/microsoft/vcpkg/pull/39332#pullrequestreview-2135592434

This should probably be after https://github.com/microsoft/vcpkg/pull/39498 makes CI test msquic for more targets.

### CODE REVIEW

**Comment by BillyONeal** [comment]
I killed the test run since it'll just merge conflict with #39332 anyway

**Review by nibanks — APPROVED** [review]

**Review by WangWeiLin-MV — APPROVED** [review]
The port usage tests pass with the following triplets:
* x64-windows

**Comment by BillyONeal** [comment]
Thanks for the review @WangWeiLin-MV :

File: versions/m-/msquic.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
```
Comment (IldarKhayrutdinov) [inline versions/m-/msquic.json:1]: It seems needs to delete `versions/m-/ms-quic.json`

Comment (WangWeiLin-MV) [reply versions/m-/msquic.json:1]: We need to preserve the version database to allow using older versions in manifest mode.

**Pull request merged**